packages/orbit-components/src/Timeline/TimelineStep/index.js

```diff
@@ -0,0 +1,160 @@
+
+const StatusIcon = ({ status }) => {
+  if (status === "success") return <CheckCircle color="success" />;
+  if (status === "warning") return <AlertCircle color="warning" />;
+  if (status === "critical") return <CloseCircle color="critical" />;
+
+  return <Circle color="tertiary" size="small" />;
+};
+
+const renderStatus = (status, theme) => {
+  if (status === "success") return theme.orbit.colorTextSuccess;
+  if (status === "warning") return theme.orbit.colorTextWarning;
+  if (status === "critical") return theme.orbit.colorTextCritical;
+
+  return theme.orbit.paletteCloudNormalHover;
+};
+
+const ProgressLineStyled = styled.span`
+  ${({ desktop, theme, status, isLeft }) => css`
+    ${!desktop && `position: absolute`};
+    width: ${desktop ? "50%" : "2px"};
+    ${isLeft ? `left: 0` : `right: -20px`};
+    z-index: -1;
+    ${!desktop && `left: 11px`};
+    background: ${renderStatus(status, theme)};
+    height: ${desktop ? `2px` : `calc(100% + 20px)`};
+  `}
+`;
+
+ProgressLineStyled.defaultProps = {
+  theme: defaultTheme,
+};
+
+const TimeWrapperStyled = styled.div`
+  visibility: ${({ visible }) => (visible ? `visible` : `hidden`)};
+`;
+
+const IndentStyled = styled.div`
+  padding-left: ${({ theme }) => theme.orbit.spaceXSmall};
+`;
+
+IndentStyled.defaultProps = {
+  theme: defaultTheme,
+};
+
+const RelativeStyled = styled.div`
+  position: relative;
+  ${({ inner, theme }) =>
+    inner &&
+    css`
+      width: calc(100% + ${parseInt(theme.orbit.spaceSmall, 10) * 2}px);
+      display: flex;
+      align-items: center;
+    `};
+`;
+
+RelativeStyled.defaultProps = {
+  theme: defaultTheme,
+};
+
+const DescriptionStyled = styled.div`
+  max-width: 250px;
+  width: 100%;
+`;
+
+const TimelineStep = ({ children, step, time, status }: Props) => {
+  const { statuses, setStatuses } = useStatuses();
+  const { index, last } = useStep();
+  const { isDesktop } = useMediaQuery();
+
+  const nextStatus = statuses[index + 1];
+
+  React.useEffect(() => {
+    setStatuses(prev => ({ ...prev, [index]: status || null }));
+  }, [setStatuses, status, index]);
+
+  return isDesktop ? (
+    <RelativeStyled>
+      <Stack inline direction="column" align="center" spacing="compact" spaceAfter="large">
+        <TimeWrapperStyled visible={!!status}>
+          <Text size="small">{time}</Text>
+        </TimeWrapperStyled>
+        <RelativeStyled inner>
+          <ProgressLineStyled data-test="progressLine" desktop isLeft status={status} />
+          <IconWrapperStyled>
+            <StatusIcon status={status} />
+          </IconWrapperStyled>
+          <ProgressLineStyled
+            data-test="progressLine"
+            desktop
+            status={nextStatus || (last && status)}
+          />
+        </RelativeStyled>
+        <Stack flex align="center" direction="column">
+          <Stack inline spacing="condensed" align="center">
+            <Badge type={status}>
+              <Text type={status || "secondary"}>{step}</Text>
```

Comment:
We can use `Text` component here (also in the mobile version). It has totally different color and line height, moreover the used size and weight is wrong.

I would use only the `Badge` component in these cases. To implement `secondary` color for the text we can:
- either use `BadgePrimitive` to recreate `neutral` type of the `Badge` and just tweak the color,
- or use `neutral` badge, and overwrite its color to `secondary`.

With both approaches it will look like in the design specification.

Reply:
should  i make changes to neutral badge or it's gonna be discussed with Will first?

Reply:
No, as I mentioned, please either use `BadgePrimitive` to **recreate** the type `neutral` of the `Badge` component and just tweak the font color to match the design specification.

Or, alternatively, you can use `<Badge type="secondary" />` and just overwrite the font color locally by selector in your styled component.

Reply:
aha, got it, thanks :)